src/subscription_manager/cli_command/abstract_syspurpose.py

```diff
@@ -0,0 +1,403 @@
+                self._print_valid_values(valid_fields)
+            else:
+                print(_('No valid values provided for "{syspurpose_attr}"').format(syspurpose_attr=self.attr))
+        else:
+            print(_('Unable to get list of valid values for "{syspurpose_attr}"').format(syspurpose_attr=self.attr))
+
+    def sync(self):
+        return syspurposelib.SyspurposeSyncActionCommand().perform(include_result=True)[1]
+
+    def _do_command(self):
+        self._validate_options()
+
+        self.cp = None
+        try:
+            # If we have a username/password, we're going to use that, otherwise
+            # we'll use the identity certificate. We already know one or the other
+            # exists:
+            if self.options.token:
+                try:
+                    self.cp = self.cp_provider.get_keycloak_auth_cp(self.options.token)
+                except Exception as err:
+                    log.error('unable to connect to candlepin server using token: "%s", err: %s' %
+                              (self.options.token, err))
+                    print(_("Unable to connect to server using token"))
+            elif self.options.username and self.options.password:
+                self.cp_provider.set_user_pass(self.options.username, self.options.password)
+                self.cp = self.cp_provider.get_basic_auth_cp()
+            else:
+                # get an UEP as consumer
+                if self.is_registered():
+                    self.cp = self.cp_provider.get_consumer_auth_cp()
+        except connection.RestlibException as err:
+            log.exception(re)
+            if getattr(self.options, 'list', None):
+                log.error("Error: Unable to retrieve %s from server: %s" % (self.attr, err))
+                system_exit(os.EX_SOFTWARE, err.msg)
+            else:
+                log.debug("Error: Unable to retrieve %s from server: %s" % (self.attr, err))
+        except Exception as err:
+            log.debug("Error: Unable to retrieve %s from server: %s" % (self.attr, err))
+
+        self.store = SyncedStore(uep=self.cp, consumer_uuid=self.identity.uuid)
+
+        if getattr(self.options, 'unset', None):
+            self.unset()
+        elif getattr(self.options, 'set', None):
+            self.set()
+        elif hasattr(self.options, 'to_add') and len(self.options.to_add) > 0:
+            self.add()
+        elif hasattr(self.options, 'to_remove') and len(self.options.to_remove) > 0:
+            self.remove()
+        elif getattr(self.options, 'list', None):
+            self.list()
+        elif getattr(self.options, 'show', None):
+            self.show()
+        else:
+            self.show()
+
+    def check_syspurpose_support(self, attr):
+        if self.is_registered() and not self.cp.has_capability('syspurpose'):
+            print(_("Note: The currently configured entitlement server does not support System Purpose {attr}.".format(
+                attr=attr)))
```

Comment:
Using both %s and .format() to build strings.   I personally think this should be consistent.